adhocracy4/filters/filters.py

```diff
@@ -75,6 +78,27 @@ def filter(self, qs, value):
         return qs.order_by(*ordering)
 
 
+class DynamicChoicesOrderingFilter(DynamicChoicesMixin,
+                                   DistinctOrderingFilter):
+
+    def __init__(self, *args, **kwargs):
+        if 'widget' not in kwargs:
+            kwargs['widget'] = OrderingWidget
+        kwargs['empty_label'] = None
+        super().__init__(*args, **kwargs)
+
+    def filter(self, qs, value):
+        qs = qs.annotate_positive_rating_count() \
+            .annotate_negative_rating_count() \
+            .annotate_comment_count()
+        if value == ['-comment_count']:
+            return qs.order_by('-comment_count')
+        elif value == ['-positive_rating_count']:
+            return qs.order_by('-positive_rating_count')
+
```

Comment:
Not sure, but maybe do we want a short comment, why we are annotating here? Just so we remember in the future..

Reply:
Yes! Good idea!